gpcontrib/arenadata_toolkit/arenadata_toolkit.c

```diff
@@ -1,4 +1,188 @@
+#include <dirent.h>
+#include <sys/stat.h>
+#include <ctype.h>
+
 #include "postgres.h"
 #include "fmgr.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "storage/fd.h"
+#include "nodes/execnodes.h"
+#include "cdb/cdbvars.h"
+#include "utils/builtins.h"
+#include "utils/relfilenodemap.h"
+#include "utils/timestamp.h"
+#include "catalog/pg_tablespace.h"
 
 PG_MODULE_MAGIC;
+
+typedef struct
+{
+	char     *datpath;
+	DIR      *dirdesc;
+	TupleDesc tupdesc;
+} user_fctx_data;
+
+static int should_skip_file(struct dirent *direntry)
```

Comment:
How about such an API?
```c
static int should_skip_file(const char *filename)
```
Although the check for the directory will have to be removed from this function.